removes previous lack of HUD implants

SealCure opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You'll find a README and example file in .\html\changelogs\ for further instructions.

You can also find a template for adding your changelog directly to the PR description here: https://github.com/Baystation12/Baystation12/wiki/Automatic-changelog-generation
-->
:cl:
rscadd: Adds two new implants: a medHUD and a secHUD, toggleable with an action button. Both can be implanted into normal and prosthetic heads.
/:cl:
Everyone likes implants, right? With the addition of these two new implantable HUDs, now you have a new reason to head to the roboticist. Both require the same amount of research from science as normal HUDs and are printable from the robotics mechfab.
Now with an action button toggle, so you can stop seeing HUD icons even when you close your eyes.

MuckerMayhem commented on code/modules/augment/passive/hudimplants.dm:
```diff
@@ -0,0 +1,36 @@
+	
+	if(!owner)
+		return
+
+	if(HUD_MEDICAL)
+		req_access = list(access_medical)
```
```suggestion

	if (hud_type == HUD_MEDICAL)
		req_access = list(access_medical)
```

MuckerMayhem commented on code/modules/augment/passive/hudimplants.dm:
```diff
@@ -0,0 +1,36 @@
+/obj/item/organ/internal/augment/hud/Process()
+	..()
+	
+	if(!owner)
```
```suggestion
	if (!owner)
```

MuckerMayhem commented on code/modules/augment/passive/hudimplants.dm:
```diff
@@ -0,0 +1,36 @@
+		req_access = list(access_medical)
+		if(allowed(owner))
+			process_med_hud(owner, 1)
+	
+	if(HUD_SECURITY)
+		req_access = list(access_security)
```
```suggestion
	
	if (hud_type == HUD_SECURITY)
		req_access = list(access_security)
```

MuckerMayhem commented on code/modules/augment/passive/hudimplants.dm:
```diff
@@ -0,0 +1,36 @@
+	
+	if(HUD_SECURITY)
+		req_access = list(access_security)
+		if(allowed(owner))
```
```suggestion
		if (allowed(owner))
```

SealCure commented:
Dreamchecker no longer screams at me, will continue wrapping my head around the rest tomorrow

SealCure commented:
Finally ready

MuckerMayhem commented on code/modules/augment/active/hudimplants.dm:
```diff
@@ -0,0 +1,40 @@
+		if (allowed(owner))
+			process_sec_hud(owner, 1)
+
+/obj/item/organ/internal/augment/active/hud/activate()
+	if (!can_activate())
+		return
+	else if (active)
+		active = FALSE
+	else if (!active)
+		active = TRUE
```
```suggestion
/obj/item/organ/internal/augment/active/hud/activate()
	if (!can_activate())
		return
	active = !active
```

SealCure replied:
Definitely a lot cleaner than what I was doing

MuckerMayhem commented on code/modules/augment/active/hudimplants.dm:
```diff
@@ -0,0 +1,40 @@
+	if (!owner)
+		return
+
+	else if (hud_type == HUD_MEDICAL && active)
+		req_access = list(access_medical)
+		if (allowed(owner))
+			process_med_hud(owner, 1)
+	else if (hud_type == HUD_SECURITY && active)
+		req_access = list(access_security)
+		if (allowed(owner))
+			process_sec_hud(owner, 1)
```
```suggestion
	if (active)
		if (hud_type == HUD_MEDICAL)
			req_access = list(access_medical)
			if (allowed(owner))
				process_med_hud(owner, 1)
		else if (hud_type == HUD_SECURITY)
			req_access = list(access_security)
			if (allowed(owner))
				process_sec_hud(owner, 1)
```
No need to make it an `else if` here since if `owner` is false, it will return anyway.

SealCure replied:
Also much cleaner

MuckerMayhem commented on code/modules/augment/active/hudimplants.dm:
```diff
@@ -0,0 +1,38 @@
+		return
+
+	if(active)
+		if (hud_type == HUD_MEDICAL && active)
```
```suggestion
		if (hud_type == HUD_MEDICAL)
```

MuckerMayhem commented on code/modules/augment/active/hudimplants.dm:
```diff
@@ -0,0 +1,38 @@
+			req_access = list(access_medical)
+			if (allowed(owner))
+				process_med_hud(owner, 1)
+		else if (hud_type == HUD_SECURITY && active)
```
```suggestion
		else if (hud_type == HUD_SECURITY)
```

the `if (active)` above already covers the things being active or not, having another check for `active` inside that is redundant.

MuckerMayhem commented on code/modules/augment/active/hudimplants.dm:
```diff
@@ -0,0 +1,38 @@
+	if (!owner)
+		return
+
+	if(active)
```
```suggestion
	if (active)
```
styling

Spookerton commented:
proviso that this may need a pass to add emp behavior, idk

SierraKomodo commented:
If it needs EMP behavior added I think I'd prefer it being part of the original PR that introduces the items instead of being a nebulous TODO that'll probably never get handled afterward.

SealCure commented:
Everyone's favourite EMP blindness added

SealCure committed:
Add implantable HUDs

Made printable
Spelling
Implement suggestions
Repath
Finally
Implement fix


Redundant checks


Add emp behaviour


Style
